pkg/kubelet/network/hostport/hostport_syncer.go

```diff
@@ -213,6 +216,33 @@ func (h *hostportSyncer) SyncHostports(natInterfaceName string, activePodPortMap
 	// Accumulate NAT chains to keep.
 	activeNATChains := map[utiliptables.Chain]bool{} // use a map as a set
 
+	// Populate the LOCALDEST chain.
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return err
+	}
+	// Only consider packets with dest of a real interface addr for hostports.
+	for _, a := range addrs {
+		astr := a.String() // CIDR notation
+		ip, ipnet, err := net.ParseCIDR(astr)
+		if err != nil {
+			return err
+		}
+		if ip.To4() != nil {
+			ipstr := ip.String()
+			if ip.IsLoopback() {
+				// Special case loopback devices.
+				ipstr = ipnet.String()
+			}
+			args := []string{
+				"-A", string(kubeLocalDestChain),
+				"-m", "comment", "--comment", `"maybe kube hostport"`,
+				"-d", ipstr,
+				"-j", string(kubeHostportsChain),
```

Comment:
Same here. `ensureKubeHostportChainLinked` need to happen before this. Otherwise, KUBE-HOSTPORTS may not exist and it will fail at iptables-restore.